File: regexlib/python_re2_test_file/regexlib_212.py
# 212
# ^[\n &lt;&quot;']*([a-zA-Z0-9._-]+@[a-zA-Z0-9._-]+)
# POLYNOMIAL
# nums:5
# POLYNOMIAL AttackString:""+"l"*10000+"!1 __POA(i)"

import re2 as re
from time import perf_counter

regex = """^[\n &lt;&quot;']*([a-zA-Z0-9._-]+@[a-zA-Z0-9._-]+)"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "" + "l" * i * 10000 + "!1 __POA(i)"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *10000}: took {DURATION} seconds!")

File: regexlib/javascript_test_file/regexlib_212.js
/* 212
 * ^[\n &lt;&quot;']*([a-zA-Z0-9._-]+@[a-zA-Z0-9._-]+)
 * POLYNOMIAL
 * nums:3
 * POLYNOMIAL AttackString:""+"l"*10000+"!1 __POA(i)"
 */
var REGEX = /^[\n &lt;&quot;']*([a-zA-Z0-9._-]+@[a-zA-Z0-9._-]+)/
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = '' + 'l'.repeat(i*10000) + '!1 __POA(i)'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 10000 + ": " + time_cost+" ms")
}